networkx/drawing/layout.py

```diff
@@ -403,6 +404,12 @@ def fruchterman_reingold_layout(
         The edge attribute that holds the numerical value used for
         the edge weight.  If None, then all edge weights are 1.
 
+    node_weight : string or None   optional (default= None)
+        The node attribute that holds the numerical value used for
+        the node weight.  If None, then all node weights are 1. If
+        the graph is large (>500 nodes), a non-None value is not
+        recommended for reasons of computational efficiency.
```

Comment:
This last sentence may fit better in the Notes section, unless there is a significant difference in speed/memory consumption between the original and "node-weighted" FR functions.

Reply:
To be honest the only reason I've had to make note of this at all is because I don't want to touch the `_sparse_fruchterman_reingold` function since I don't really understand how it is much more efficient than `_fruchterman_reingold` even in the sparse case. Due to the anti-gravity force used in a Fruchterman-Reingold layout it still has to iterate through all possible pairs of nodes... so I don't see how the sparse nature of *spring* force could have an impact on the order of the time complexity.

Reply:
I don't think the sparse code improves the time complexity. It saves on RAM. At least that's what I recall. There is also a comment in the code about how the speedy code is still O(V^2). So I think the time complexity is that for all versions.